docs/API.md

```diff
@@ -437,8 +411,82 @@ Options
 There's a general method of setting options on a socket in the SRT C API, similar
 to the system setsockopt/getsockopt functions.
 
-Synopsis
---------
+Types used in socket options
+----------------------------
+
+Possible types of options are:
+
+* `int32_t` - usually you can treat it as `int` equivalent, as this type
+wasn't changed the size on 64-bit systems, although the definition is using
+`int32_t` type so that the bit size is fixed by definition. In some cases
+the value is expressed using an enumeration type (see below).
+
+* `int64_t` - some options need the paramter specified as 64-bit integer
+
+* `bool` - You need to use a boolean type, the builtin one in C++ and the
+one from `<stdbool.h>` for C, although when setting an option, passing the
+value through `int` type is also tolerated and properly recognized. When
+getting an option, however, you better use the `bool` type, although you
+can risk passing a variable of `int` type initialized with 0 and then check if
+the resulting value is equal to 0 (just don't compare the result with 1).
+
+* `string` - When setting, you pass the character array pointer as value
+and the string length as length. When getting, you have to pass a long
+enough array (with specified size in the size variable). Every option with
+this type that can be read should specify the maximum length of that array.
+
+
+Enumeration types used in options
+---------------------------------
+
+
+### 1. `SRT_TRANSTYPE`. Used by `SRTO_TRANSTYPE` option:
+
+* `SRTT_LIVE`: Live mode.
+* `SRTT_FILE`: File mode.
+
+See below [Transmission types](#transmission-types) for details.
+
+
+### 2. `SRT_KM_STATE`. This defined encryption state as performed by the Key
+Material Exchange, used by `SRTO_RCVKMSTATE`, `SRTO_SNDKMSTATE` and
+`SRTO_KMSTATE` options:
+
+- `SRT_KM_S_UNSECURED`: no encryption/descryption. If this state is only on
+the receiver, received encrypted packets will be dropped.
+
+- `SRT_KM_S_SECURING`: pending security (HSv4 only). This is a temporary state
+used only if the connection is made has HSv4 and the Key Material Exchange is
+not finished yet. On HSv5 not possible because Key Material Exchange for the
+initial key is done in the handshake.
```

Comment:
```suggestion
used only if the connection uses HSv4 and the Key Material Exchange is
not finished yet. On HSv5 this is not possible because the Key Material Exchange for the
initial key is done in the handshake.
```